### PR TITLE
Add verbosity to TVAE (progress bar + save the loss values)

### LINKED ISSUE #300: Add verbosity TVAE (progress bar + save the loss values)
### Problem Description
Similar to #298, we should also add a verbosity option for TVAE that shows the user the fitting progress per epoch. After fitting, it should be possible to see the loss values.

### Expected behavior
1. Add a `verbose` parameter to TVAE (just like CTGAN). By default it can be `False`
2. When set to `True`, increment a progress bar per (epoch x data iteration). Also show/refresh the loss value, similar to #298 (except there is only 1 loss value)
3. Always save the loss values that correspond to the last fitting iteration. The columns can now be `Epoch`, `Batch` and `Loss`.

```python
>>> model = TVAE(verbose=True, epochs=100)
>>> model.fit()
Fitting TVAE ...
Loss 0.024 |████████████████████████████         | 750/1000 [00:33<00:10, 229.00it/s]

# after modeling
>>> model.loss_values
   Epoch     Batch    Loss 
0      1     1        1.7863
1      1     2        1.5484
2      1     3        1.3633
```

### PR DESCRIPTION
**Pull request by frances-h**
Resolve #300

### CODE REVIEW

**Review by rwedge — COMMENTED** [review]
Should we test verbose=True?

File: ctgan/synthesizers/tvae.py [hunk 1]
```diff
@@ -176,6 +188,26 @@ def fit(self, train_data, discrete_columns=()):
...
+                'Epoch': [i] * len(batch),
+                'Batch': batch,
+                'Loss': loss_values
+            })
+            if not self.loss_values.empty:
+                self.loss_values = pd.concat(
+                    [self.loss_values, epoch_loss_df]
+                ).reset_index(drop=True)
+            else:
+                self.loss_values = epoch_loss_df
```
Comment (amontanez24) [inline ctgan/synthesizers/tvae.py:204]: Since you made the skeleton of the dataframe above can't you just always concatenate?

File: ctgan/synthesizers/tvae.py [hunk 2]
```diff
@@ -176,6 +188,26 @@ def fit(self, train_data, discrete_columns=()):
...
+
+            if self.verbose:
+                iterator.set_description(
+                    iterator_description.format(
+                        loss=loss.detach().cpu().item()))
```
Comment (amontanez24) [inline ctgan/synthesizers/tvae.py:209]: can we add an integration test that checks the value is printed in the right format?

File: ctgan/synthesizers/tvae.py (hunk 1 above)
Comment (frances-h) [reply ctgan/synthesizers/tvae.py:204]: Pandas throws a `FutureWarning` when you try to concat with an empty dataframe
```bash
CTGAN/ctgan/synthesizers/tvae.py:201: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  self.loss_values = pd.concat(
```

**Review by rwedge — APPROVED** [review]

File: ctgan/synthesizers/tvae.py (hunk 2 above)
Comment (amontanez24) [reply ctgan/synthesizers/tvae.py:210]: Sorry I meant an integration test without mocks that used something like [capsys](https://docs.pytest.org/en/7.4.x/how-to/capture-stdout-stderr.html#accessing-captured-output-from-a-test-function) to check the printed output

**Review by amontanez24 — APPROVED** [review]
LGTM! Thanks for adding the tests!
